.github/workflows/build.yml

```diff
@@ -0,0 +1,32 @@
+name: build
+
+on:
+  release:
+    types: [ released ]
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  check:
+    if: github.event_name == 'release'
+    name: check that `requirements-sdp.txt` is populated
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - run: grep -v '^ *#' requirements-sdp.txt
+  build:
+    needs: [ check ]
+    if: always() && (needs.check.result == 'success' || needs.check.result == 'skipped')
```

Comment:
I don't grok this condition but whatever works for you, I guess.

Reply:
this allows the `requirements-sdp.txt` check job to be skipped (for a PR or manual dispatch) but still trigger the `build` job

Reply:
I have no idea what that requirements file is for, so I'll leave that to your good judgement.